content/articles/introduction-to-c/index.md

```diff
@@ -0,0 +1,202 @@
+but this series will explain everything from ground level to high level
+
+### Table of contents
+
+- History of C Program
+- Structure of C program
+- Installing C in our System
+- Writing the First C program
+- Files used in C program
+- Uses Of C programming
+
+### History of C Program
+
+C is a practical programming language. In the 1970's "C" was developed by Dennis Ritchie and used by the UNIX operating system at Bell Laboratories.
+
+- Why this language named C?
+  - Before "C" there is a language called "B" and Some of the Options are derived from "B"
+- What is the purpose of the C language?
+  - Early "C" is only used to design and implement the system software. But later some features were add to "C" then it is used to develop portable application software too.
+
+
+### Structure of C program
+
+`C` program is structured of preprocessor commands, a global declaration section, and one or more functions
+
+```
+preprocessor directives
+
+Global declarations
+main()
+{
+ Local declarations
+ Statements
+ }
+ Function 1()
+{
+ Local declarations
+ Statements
+ }
+ Function N()
+{
+ Local declarations
+ Statements
+ }
+```
+
+- preprocessor directives
+
+  - Preprocessor directives are lines included in the code of programs preceded by a hash sign `#`
+  - One of the most important and commonly used preprocessor commands is `include` which tells the compiler to execute the program
+
+- Functions in C
+  - `C` program contains one or more functions.
+  - A group of `C` statements and that statements executed together is known as Function 
+  - `main()` is the most important function and is a part of every `C` program
+
+### Installing C in our System
+
+- In Windows
+
+  - Download [Dev-C++](https://sourceforge.net/projects/orwelldevcpp/) from your favourite browser
+  - Open the folder where you downloaded
+  - And Install as Normal Application
+
+- In Linux
+
+  - Open a Terminal window. Enter the command this command to install build-essential packages
+    `$ Sudo apt-get install build-essential`
+
+- In Mac
+  - Open a Terminal window. Enter the command `clang --version`
+  - If you want to install or update the Clang compilers, enter the command command `xcode-select --install`
+  - You will get a pop up select `INSTALL` and ` Agree`
+
+### Writing the First C program
+
+- In windows
+
+  - Open `Start Menu > All Programs > Bloodshed Dev-C++ > Dev-C++`
+  - Then Navigate to `File > New > Source File`.It will navigate to empty space or file where we can write our code 
+  
+
+  ```
+      #include <stdio.h>
+       int main()
+       {
+        printf("Welcome to Introduction to C programming");
+        return 0;
+       }
+  ```
+
+  - Navigate `File > Save As` to save your file. The select location you want to save and save the source file (the file which you written the code)by changing file type as `C` with the extension of `(*.c)`.
+
+  #### To Run the code
```

Comment:
Remove extra space before the title. Check the entire article for formatting issues

Reply:
done